### ISSUE TITLE
False Positive on CKV2_AWS_12: "Ensure the default security group of every VPC restricts all traffic"

### ISSUE THREAD

**Issue by txsutton** [bridgecrewio/checkov#5097]
**Describe the issue**
I get the error when I try and create a VPC even with the aws_default_security_group added

**Examples**
```
resource "aws_vpc" "spoke_vpc" {
  ipv4_ipam_pool_id                = data.aws_vpc_ipam_pool.pool_eu.id
  cidr_block                       = local.spoke_vpc.vpc_cidr
  enable_dns_support               = true
  enable_dns_hostnames             = true
  assign_generated_ipv6_cidr_block = false
  tags = {
    Name = local.spoke_vpc.name
  }
}

resource "aws_default_security_group" "spoke_vpc" {
  vpc_id = aws_vpc.spoke_vpc.id

  ingress {
    protocol  = "-1"
    self      = true
    from_port = 0
    to_port   = 0
  }

  egress {
    from_port   = 0
    to_port     = 0
    protocol    = "-1"
    cidr_blocks = ["0.0.0.0/0"]
  }
}
```

**Version (please complete the following information):**
2.3.238

**Additional context**
If I put a hyphen in front of the ingress word then the error goes away

**Comment by tarfeef101** [comment]
the security group config you've linked is completely permissive, so the check is valid (it is telling you to restrict all traffic with the default security group, so you're forced to explicitly create one for every resource)

**Comment by txsutton** [comment]
The code provided is a copy of the Fix listed here: https://docs.bridgecrew.io/docs/networking_4, can you provide a working resource that is both secure and compliant with checkov?

**Comment by gruebel** [comment]
as in the docs written, you just need to remove the ingress and egress blocks inside the resource
```hcl
resource "aws_vpc" "spoke_vpc" {
  ipv4_ipam_pool_id                = data.aws_vpc_ipam_pool.pool_eu.id
  cidr_block                       = local.spoke_vpc.vpc_cidr
  enable_dns_support               = true
  enable_dns_hostnames             = true
  assign_generated_ipv6_cidr_block = false
  tags = {
    Name = local.spoke_vpc.name
  }
}

resource "aws_default_security_group" "spoke_vpc" {
  vpc_id = aws_vpc.spoke_vpc.id
}
```
